Week1/LaurAlejandraC/unit-test/src/test/java/com/globant/bootcamp/spring/unittest/Scenario1Test.java

```diff
@@ -0,0 +1,40 @@
+package com.globant.bootcamp.spring.unittest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import springbootcamp.unittest.kata.business.AccountServiceImpl;
+import springbootcamp.unittest.kata.data.AccountRepository;
+import springbootcamp.unittest.kata.data.AccountRepositoryImpl;
+import springbootcamp.unittest.kata.domain.Account;
+import springbootcamp.unittest.kata.domain.AccountType;
+import springbootcamp.unittest.kata.domain.TransferResponse;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Scenario1Test {
+    private AccountRepository accountRepository;
+    private Account currentAccount;
+    private Account savingsAccount;
+
+    @Before
+    public void setUp(){
+        currentAccount = new Account("User", 3000, AccountType.CURRENT);
+        savingsAccount = new Account("User", 1500, AccountType.SAVING);
+        accountRepository = new AccountRepositoryImpl(new ArrayList<>(Arrays.asList(currentAccount, savingsAccount)));
+    }
+
+    @Test
+    public void transfer(){
+        AccountServiceImpl accountService = new AccountServiceImpl(accountRepository);
+
+        try{
+            TransferResponse transferResponse = accountService.transferMoney(currentAccount, savingsAccount, 2500);
+            Assert.assertTrue(transferResponse.getCurrentAccount().getBalance() == 500 && transferResponse.getSavingAccount().getBalance() == 4000);
+        }
+        catch (Exception ex){
```

Comment:
You shouldn't have a try - catch structure in a unit test (usually). Your unit test should be focused on one flow, it means, if you have one flow where an exception is expected, you should create a different unit test with that flow. You can use @Test(throw=exception) to validate that an exception was thrown, or for more advanced cases, @Rule and ExpectedException class. It is fine to declare the exception in the sign method using throws word if for that flow doesn't matter the exception.

Reply:
I agree, I know that you have made another test case to check the exception and that was I expected, so please remove the "try/catch" clause and that's it :)